kolibri/content/views.py

```diff
@@ -58,3 +58,31 @@ def get(self, request, zipped_filename, embedded_filepath):
         response["Accept-Ranges"] = "none"
 
         return response
+
+class DownloadContentView(View):
+
+    def get(self, request, filename, new_filename):
+        """
+        Handles GET requests and serves a static file as an attachment.
+        """
+
+        # calculate the local file path of the file
+        path = get_content_storage_file_path(filename)
+
+        # if the file does not exist on disk, return a 404
+        if not os.path.exists(path):
+            raise Http404('"%(filename)s" does not exist locally' % {'filename': filename})
+
+        # generate a file response
+        response = FileResponse(open(path, 'rb'))
+
+        # set the content-type
+        response['Content-Type'] = mimetypes.guess_type(new_filename)[0]
+
+        # set the content-disposition as attachment to force download
+        response['Content-Disposition'] = 'attachment; filename=%s' % new_filename
```

Comment:
Yeah, you're doing exactly what I was recommending. But when I looked at the code, I got wondering about encodings, escaping characters, unicode, etc -- and searched and found that link. It sounds like there are a bunch of possible browser differences/issues around the edge cases. If we can just leave off the server-side `filename=` piece, leaving the `attachment` part, browsers will still use the right filename, since it's the last component of the URL. That seems to be what some people are recommending.